proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,321 @@
+
+##### Case 2 for Project to Project reference
+
+`ProjectReference` in parent `ClassLibrary1` csproj file:
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\refissue\refissue.csproj" PrivateAssets="none" ExcludedAssetsFlow="true"/>
+  </ItemGroup>
+```
+
+And `PackageReference` in `refissue.csproj` file:
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <VersionSuffix>beta</VersionSuffix>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.138-beta" PrivateAssets="build"/>
+  </ItemGroup>
+```
+
+Before changing, `Microsoft.Windows.CsWin32.props` file in `build` asset doesn't flow to `ClassLibrary1.csproj.nuget.g.props` file in `obj` folder.
+
+After change, same as before.
+
+##### Case 3 for Project to Project reference
+
+`ProjectReference` in grandparent `Top` csproj file:
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\intermediate\intermediate.csproj" />
+  </ItemGroup>
+```
+
+`ProjectReference` in parent `intermediate` csproj file:
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\refissue\refissue.csproj" />
+  </ItemGroup>
+```
+
+And `PackageReference` in `refissue.csproj` file:
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <VersionSuffix>beta</VersionSuffix>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.138-beta"  PrivateAssets="compile" IncludeAssets="none" ExcludedAssetsFlow="true" />
+  </ItemGroup>
+```
+
+Before changing, `Microsoft.Windows.CsWin32.props` file in `build` asset doesn't flow to `intermediate.csproj.nuget.g.props` file in `intermediate/obj` folder or `top.csproj.nuget.g.props` file in `top/obj` folder.
+
+After change, `Microsoft.Windows.CsWin32.props` file in `build` asset flow to `intermediate.csproj.nuget.g.props` file in `intermediate/obj` folder, but doesn't flow to `top.csproj.nuget.g.props` file in `top/obj` folder.
+
+### Technical explanation
+
+We already have a [logic](https://github.com/NuGet/NuGet.Client/blob/380415d812681ebf1c8aa0bc21533d4710514fc3/src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs#L577-L582) that combines `IncludeAssets/ExcludeAssets` options with `PrivateAssets` for pack and restore operation, we just make logic depending on opt-in metadata at package reference.
```

Comment:
@nkolev92 
Do you know if have something special for `ProjectReference` than `PackageReference` asset calculation? If yes could you  please give me code pointers?

Reply:
They're treated the same.